platform/commonUI/edit/src/actions/SaveAsAction.js

```diff
@@ -0,0 +1,169 @@
+            $injector,
+            policyService,
+            dialogService,
+            creationService,
+            copyService,
+            context
+        ) {
+            this.domainObject = (context || {}).domainObject;
+            this.injectObjectService = function(){
+                this.objectService = $injector.get("objectService");
+            };
+            this.policyService = policyService;
+            this.dialogService = dialogService;
+            this.creationService = creationService;
+            this.copyService = copyService;
+        }
+
+        /**
+         * @private
+         */
+        SaveAsAction.prototype.createWizard = function (parent) {
+            return new CreateWizard(
+                this.domainObject,
+                parent,
+                this.policyService
+            );
+        };
+
+        /**
+         * @private
+         */
+        SaveAsAction.prototype.getObjectService = function(){
+            // Lazily acquire object service (avoids cyclical dependency)
+            if (!this.objectService) {
+                this.injectObjectService();
+            }
+            return this.objectService;
+        };
+
+        function resolveWith(object){
+            return function () {
+                return object;
+            };
+        }
+
+        /**
+         * Save changes and conclude editing.
+         *
+         * @returns {Promise} a promise that will be fulfilled when
+         *          cancellation has completed
+         * @memberof platform/commonUI/edit.SaveAction#
+         */
+        SaveAsAction.prototype.perform = function () {
+            // Discard the current root view (which will be the editing
+            // UI, which will have been pushed atop the Browse UI.)
+            function returnToBrowse(object) {
+                if (object) {
+                    object.getCapability("action").perform("navigate");
+                }
+                return object;
+            }
+
+            return this.save().then(returnToBrowse);
+        };
+
+        /**
+         * @private
+         */
+        SaveAsAction.prototype.save = function () {
+            var self = this,
+                domainObject = this.domainObject,
+                copyService = this.copyService;
+
+            function doWizardSave(parent) {
+                var wizard = self.createWizard(parent);
+
+                return self.dialogService
+                    .getUserInput(wizard.getFormStructure(true),
+                        wizard.getInitialFormValue()
+                    ).then(wizard.populateObjectFromInput.bind(wizard));
+            }
+
+            function fetchObject(objectId){
+                return self.getObjectService().getObjects([objectId]).then(function(objects){
+                    return objects[objectId];
+                });
+            }
+
+            function getParent(object){
+                return fetchObject(object.getModel().location);
```

Comment:
Ok, so I looked into this and the short answer is no. The original object doesn't get contextualized because it doesn't have a location (because it's working with two models - the original and the cached). I could probably fix it in this branch, but I'm disinclined to in light of #824 as there is only one canonical model. I am using this very method with new objects in #824 so I know it will work there. I'll do this particular refactor in that branch once this is merged.